Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1860,16 +1883,39 @@ PhysicalParticleContainer::PushP (int lev, Real dt,
                 &ll4symtry, &WarpX::l_lower_order_in_v, &WarpX::do_nodal,
                 &lvect_fieldgathe, &WarpX::field_gathering_algo);
 
-            warpx_particle_pusher_momenta(&np,
-                                          m_xp[thread_num].dataPtr(),
-                                          m_yp[thread_num].dataPtr(),
-                                          m_zp[thread_num].dataPtr(),
-                                          uxp.dataPtr(), uyp.dataPtr(), uzp.dataPtr(),
-                                          m_giv[thread_num].dataPtr(),
-                                          Exp.dataPtr(), Eyp.dataPtr(), Ezp.dataPtr(),
-                                          Bxp.dataPtr(), Byp.dataPtr(), Bzp.dataPtr(),
-                                          &this->charge, &this->mass, &dt,
-                                          &WarpX::particle_pusher_algo);
+            // This wraps the momentum advance so that inheritors can modify the call.
+            // Extract pointers to the different particle quantities
+            Real* AMREX_RESTRICT gi = m_giv[thread_num].dataPtr();
+            Real* AMREX_RESTRICT uxpp = uxp.dataPtr();
+            Real* AMREX_RESTRICT uypp = uyp.dataPtr();
+            Real* AMREX_RESTRICT uzpp = uzp.dataPtr();
+            Real* AMREX_RESTRICT Expp = Exp.dataPtr();
+            Real* AMREX_RESTRICT Eypp = Eyp.dataPtr();
+            Real* AMREX_RESTRICT Ezpp = Ezp.dataPtr();
+            Real* AMREX_RESTRICT Bxpp = Bxp.dataPtr();
+            Real* AMREX_RESTRICT Bypp = Byp.dataPtr();
+            Real* AMREX_RESTRICT Bzpp = Bzp.dataPtr();
```

Comment:
Not sure how important it is, but `gi`, `uxpp`, `uypp`, `uzpp` could be declared as 
```
Real * const AMREX_RESTRICT uxpp = uxp.dataPtr();
```
instead of 
```
Real* AMREX_RESTRICT uxpp = uxp.dataPtr();
```
and `{E,B}{x,y,z}pp` could be declared as 
```
const Real * const AMREX_RESTRICT Expp = Exp.dataPtr();
```
instead of
```
Real* AMREX_RESTRICT Expp = Exp.dataPtr();
```
Not sure it would change anything, it might help the compilers a bit.

Reply:
It's an easy change so I'll do it.